app/controllers/api_controller/parser.rb

```diff
@@ -299,5 +299,11 @@ def fetch_action_hash(aspec, method_name, action_name)
     def collection_option?(option)
       collection_config.option?(@req.collection, option) if @req.collection
     end
+
+    def assert_id_not_specified(data, type)
```

Comment:
Funnily enough I think we are *already* doing this check somewhere else...perhaps in the manager? I was going to write some tests and investigate this week....but I think this could be a straight-up deletion